### PR TITLE
refactor: convert mersenne into a class

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Created in relation to #280.

Converted `MersenneTwister19937` function to a js class declaration.

Things I want to change but need approval:
- proper camelcase for all methods and variables/properties
- remove the file scoped variable `dbg` in line 54 since it is not used anywhere
- property jsdocs for public methods

### CODE REVIEW

**Comment by xDivisionByZerox** [comment]
@Shinigami92 can you confirm that `dbg` in line 54 is unused and can be removed?
It gets assigned in line 205 but doesn't do anything.
I have a second branch on my fork where I removed it and all test still work fine.

**Comment by xDivisionByZerox** [comment]
I guess I fked up the method scoping (?). I'm gonna investigate this.

**Comment by Shinigami92** [comment]
> dbg

Yes, it could be that this is not used, BUT maybe it is something like a global weird thing that does any magic sideeffect :grimacing: 
If not, please purge it with fire :fire:

File: src/vendor/mersenne.ts [hunk 1]
```diff
@@ -53,237 +53,293 @@
 
 let dbg: number;
 
-function MersenneTwister19937(): void {
+class MersenneTwister19937 {
```
Comment (Shinigami92) [inline src/vendor/mersenne.ts:56]: Maybe we should just use `export default` here instead of at the bottom

**Review by Shinigami92 — COMMENTED** [review]
Had a first quick look and LGTM, will approve later when I review deeper
Maybe you will also like to improve some stuff

File: src/vendor/mersenne.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply src/vendor/mersenne.ts:56]: As far as I know, it doesn't matter where you put the export default statement (unless you doing some weird side-effect call in between). But I can put it up there if you want.

**Comment by xDivisionByZerox** [comment]
> Yes, it could be that this is not used, BUT maybe it is something like a global weird thing that does any magic sideeffect 😬
If not, please purge it with fire 🔥

That's why I was asking if you could checkout [my branch](https://github.com/xDivisionByZerox/faker/tree/test/remove-local-scoped-variable) where I removed it since on my machine it works fine after removing. Or is there a way to trigger the CI pipeline manually on a specific branch?

File: src/vendor/mersenne.ts (hunk 1 above)
Comment (Shinigami92) [reply src/vendor/mersenne.ts:56]: Yeah, I know, it's just a preference of style from my side

**Comment by Shinigami92** [comment]
> Or is there a way to trigger the CI pipeline manually on a specific branch?

The CI pipeline will be triggered when you open a PR. Just do that and we can close the PR if not needed.

File: src/vendor/mersenne.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply src/vendor/mersenne.ts:56]: No problem mate. Just always reasking to ensure that everybody understands each other correctly. Will move the `default export`

**Comment by xDivisionByZerox** [comment]
From the CI results in #531 I can confirm that removing the file scoped variable `dbg` does not influence the results of any tests.

Will remove it now add the removal to this PR.

File: src/vendor/mersenne.ts (hunk 1 above)
Comment (xDivisionByZerox) [reply src/vendor/mersenne.ts]: Was moved in fad5504

**Review by Shinigami92 — APPROVED** [review]
Not sure why the `codecov/patch` hits so hard here, but the rewrite looks good to me :shrug:

**Review by ST-DDT — DISMISSED** [review]

**Comment by Shinigami92** [comment]
Uhm... :eyes: 
Somehow github lost the author when I pressed the rebase button 😯



I don't want to steel credits for that.
@xDivisionByZerox Could you jump into your source and do following?:
1. Update your main branch in your fork `git switch main` && _git pull upstream_ (_[maybe GitHub can help with that](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork)_)
2. switch into this branch in your fork `git switch refactor/mersenne-twister-to-class`
3. `git pull --rebase`
4. `git rebase -i main`
5. `git push -f`

**Comment by xDivisionByZerox** [comment]
I'm currently at work. Will fix this afternoon.

**Comment by ST-DDT** [comment]
@Shinigami92 Everything looks good to me:
grafik

**Comment by xDivisionByZerox** [comment]
Cool. That worked.

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**